### ISSUE TITLE
`match_like_matches_macro` reports uncompilable suggestion for inconsistent variable bindings

### ISSUE THREAD

**Issue by GottemHams** [rust-lang/rust-clippy#17503]
### Summary

When the match arms have explicit, unused variable bindings that differ between arms, the suggestion preserves those in the `OR` construct. This doesn't actually compile, as Rust requires the same bindings to be present in all alternatives.

### Reproducer

Code:

```rust
enum MatchType {
    A(String),
    B(String, String),
    C,
}

fn matches(match_type: MatchType) -> bool {
    match match_type {
        MatchType::A(_str1) => true,
        MatchType::B(_str1, _str2) => true,
        _ => false,
    }
}

fn main() {
    matches(MatchType::A("a".to_string()));
    matches(MatchType::B("b1".to_string(), "b2".to_string()));
    matches(MatchType::C);
}

```

Current output:

```
warning: match expression looks like `matches!` macro
  --> src/main.rs:8:5
   |
 8 | /     match match_type {
 9 | |         MatchType::A(_str1) => true,
10 | |         MatchType::B(_str1, _str2) => true,
11 | |         _ => false,
12 | |     }
   | |_____^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.97.0/index.html#match_like_matches_macro
   = note: `#[warn(clippy::match_like_matches_macro)]` on by default
help: use `matches!` directly
   |
 8 -     match match_type {
 9 -         MatchType::A(_str1) => true,
10 -         MatchType::B(_str1, _str2) => true,
11 -         _ => false,
12 -     }
 8 +     matches!(match_type, MatchType::A(_str1) | MatchType::B(_str1, _str2))

```

Desired output (final line only because that's the only relevant one):

```
 8 +     matches!(match_type, MatchType::A(_) | MatchType::B(_, _))
```

Or perhaps even better:

```
 8 +     matches!(match_type, MatchType::A(..) | MatchType::B(..))
```

### Version

```text
rustc 1.97.0 (2d8144b78 2026-07-07)
binary: rustc
commit-hash: 2d8144b7880597b6e6d3dfd63a9a9efae3f533d3
commit-date: 2026-07-07
host: aarch64-apple-darwin
release: 1.97.0
LLVM version: 22.1.6
```
